treeherder/perf/alerts.py

```diff
@@ -331,3 +332,48 @@ def _glance_over_retrigger_range(self, data_points_to_retrigger: List[dict]):
         if retrigger_range < self._range_width:
             self.log.warning('Found small backfill range (of size {} instead of {})'
                              .format(retrigger_range, self._range_width))
+
+
+class IdentifyRetriggerableAlerts:
+    def __call__(self, frameworks: List[str], repositories: List[str], start_time: datetime, days_interval: timedelta) -> dict:
+        summaries_to_retrigger = self._fetch_by(
+            self._summaries_since(start_time),
+            frameworks,
+            repositories
+        )
+        return self._identify_retriggerables(summaries_to_retrigger, days_interval)
+
+    def _summaries_since(self, timestamp: datetime):
+        return (PerformanceAlertSummary.objects
+                .select_related('framework', 'repository')
+                .filter(created__gte=timestamp))
+
+    def _fetch_by(self, summaries_to_retrigger: QuerySet, frameworks: List[str], repositories: List[str]) -> QuerySet:
+        if frameworks:
+            summaries_to_retrigger = summaries_to_retrigger.filter(framework__name__in=frameworks)
+        return summaries_to_retrigger.filter(repository__name__in=repositories)
+
+    def _identify_retriggerables(self, summaries_to_retrigger: QuerySet, days_interval: timedelta) -> dict:
+        # prepare helpers
+        picker = AlertsPicker(
```

Comment:
Let's extract these 2 instantiations into the `__init__` of this class.

Reply:
Ok.